### PR TITLE
Add Address

### PR DESCRIPTION
**Pull request by autotelik**
Add address migration, model, form, view;  …
Update SR Permit seed codes so they contain number too
Link address data to Contact
Better factories, states jumpers
Complete address and company number

### CODE REVIEW

**Comment by autotelik** [comment]
@lewispb  After merging you may want to re-seed your DB to pick up changes to code (now code & number)

**Comment by lewispb** [comment]
cool will do, want to just run robocop over this as well?

File: config/initializers/datashift_journey.rb [hunk 1]
```diff
@@ -7,14 +7,17 @@
...
+
+    config.state_jumper_states = [
+      klass.new(:account_type, "Acc type", :application),
+      klass.new(:operator_name, "Op Name", :individual_operator),
+      klass.new(:operator_email, "Op Email", :individual_operator),
+      klass.new(:operator_telephone, "Op Tel", :individual_operator),
+      klass.new(:choose_permit, "Comp Permit", :company_permit),
+      klass.new(:declaration, "Indvl Dec", :individual_declaration),
+      klass.new(:declaration, "Comp Dec", :company_declaration)
+    ]
```
Comment (lewispb) [inline config/initializers/datashift_journey.rb]: It's not clear to me what's going on here? Perhaps better to send over an array of hashes, and let the gem handle conversion of that config into `Struct::StateJumperEntry` objects?

Comment (autotelik) [inline config/initializers/datashift_journey.rb]: each row requires 3 parameters, so I switched from a  hash to a PRO,  its just constructing a class as normal 0 but to make it more readable I stored the class Struct::StateJumperEntry in a variable .

**Comment by autotelik** [comment]
rubocop done

**Comment by lewispb** [comment]
:ship:

**Pull request merged**